### DISCUSSION TITLE
[Help] Modifying href Values

### DISCUSSION THREAD

**Discussion by BlazeFTL** [uBlockOrigin/uAssets#24691] [Q&A]
1 ) ```https://teraboxhub.net/Zb55CTQtreRQfdexNMSemG``` [NSFW]
Filters Used
```
play.engnotebook.com##center  [href], center > p, footer
ajchibatmi.com##a.download-button:others()
ajchibatmi.com##+js(nano-stb, /location.href|showButtons/, *)
```
On the last page `play.engnotebook.com` There is a Button Play In Mx Player Is It Possible To Clean The URL
`intent:https://core.mdiskplay.com/box/terabox/video/04yLfLxZj6JobQUd2ueZTg.m3u8#Intent;package=com.mxtech.videoplayer.ad;S.title=Mdiskplay;end`
To
`https://core.mdiskplay.com/box/terabox/video/04yLfLxZj6JobQUd2ueZTg.m3u8`

2 )
 ```https://sexvideos.host/en/x-video/me-and-my-girlfriend-rough-loud-moaning-amateur-sex.html?group=skim``` [NSFW]

Below The Player There Are Related Videos After Long Clicking Those Videos There Is `?group=skim` This Part...Can This Be Somehow  Removed ?

**Comment by garry-ut99** [comment]
> 1) 

Seems currently not possible, as it's similar and partially a duplicate (regarding the `#`) question/request of :
- https://github.com/uBlockOrigin/uBlock-issues/issues/2531#issuecomment-1912355907

with a difference that in your case additionally `intent:` is to be removed from the beginning of the URL.

> 2) 

Seems currently not possible with `removeparam / uritransform`, I tried this `||sexvideos.host^$doc,popup,removeparam=/^group=/` - but linter says invalid filter (`uritransform` too), nor `$doc,other` or `$all` has worked for popups, so I guess `removeparam` doesn't support cleaning links in popups. 

I ended with `||sexvideos.host^$doc,removeparam=/^group=/` but it works only if the video links are opened in the same tab or are pasted manually to the browser address bar, which doesn't solve your issue, unless you modify your browser settings to prevent opening links in new tabs so they are opened in the same tab.

(By the way I hope it gets tokenized properly, if not then maybe according to: https://github.com/uBlockOrigin/uBlock-issues/issues/760#issuecomment-727249028 one of these will be tokenized better: `group$doc,removeparam=/^group=/,domain=sexvideos.host` or this one: 
`/en/x-video/*.html?group$doc,removeparam=/^group=/,domain=sexvideos.host`)

**Reply by gorhill** [reply]
`removeparam=/^group=/` is equivalent to `removeparam=group`, there is no need to use a regex when the parameter to remove is an exact match.

**Reply by garry-ut99** [reply]
I see, initially I used `=group` , then before creating a comment I replaced my examples with `/^group=/`, because I somehow wrongly thought that `^` regex matching will match `?` in `?name=value&name2=value` so I wanted to speed it up and remove the parameter only if it occurs as the first parameter. I just temporarily forgot that that regex matching is done for separate raw `name=value` pairs and not for whole chain of values, and that `^` matches `n` and not `?` in `?name=`

**Reply by BlazeFTL** [reply]
Oh Ok...I Tried Few Combination...But None Worked...
Although I Propose Something Like
```
##+js(rpnt, href,  /**A Regex To Match(#Intent;package=com.mxtech.videoplayer.ad;S.title=Mdiskplay;end)/, , condition, [href*="mdiskplay.com"])

**Reply by garry-ut99** [reply]
You mean to have a possibility to also modify attributes's values/text and not only text between tags.
I'm not sure which scriptlet is better to have it implemented, as there are at least 3 candidates:
- `href-sanitizer`: https://github.com/gorhill/uBlock/wiki/Resources-Library#href-sanitizerjs-
- `set-attr`: https://github.com/gorhill/uBlock/wiki/Resources-Library#set-attrjs- 
- `rpnt` https://github.com/gorhill/uBlock/wiki/Resources-Library#trusted-replace-node-textjs-

If you want to only target `href` then it seems it should go into `href-sanitizer`,
unless it's meant to be universal and to support any attributes, then `set-attr` or `rpnt`,

Also `href-sanitizer` and `set-attr` don't have their separate `trusted-` versions, so if the implementation of the feature will require the scriptlet to be `trusted-`, they would require their `trusted-` versions to be created.
